pyiron_atomistics/atomistics/structure/factories/laves.py

```diff
@@ -0,0 +1,71 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+from pyiron_atomistics.atomistics.structure.factories.ase import AseFactory
+import numpy as np
+
+__author__ = "Liam Huber"
+__copyright__ = (
+    "Copyright 2021, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "1.0.1"
+__maintainer__ = "Liam Huber"
+__email__ = "huber@mpie.de"
+__status__ = "production"
+__date__ = "Jun 28, 2021"
+
+
+class LavesFactory:
+    """A collection of routines for constructing Laves phase structures."""
+
+    @staticmethod
+    def C14():
+        raise NotImplementedError
+
+    @staticmethod
+    def C15(element_a, element_b, a=None):
+        """
+        Builds a cubic $A B_2$ C15 Laves phase cell.
+
+        Example use:
+        >>> structure = C15('Al', 'Ca')
+        >>> structure.repeat(2).plot3d(view_plane=([1, 1, 0], [0, 0, -1]))
+
+        Args:
+            element_a (str): The chemical symbol for the A element.
+            element_b (str): The chemical symbol for the B element.
+            a (float): The lattice constant. (Default is None, which uses the default nearest-neighbour distance for
+                the A-type element.)
+
+        Returns:
+            (Atoms): The C15 unit cell.
+        """
+        bulk = AseFactory().bulk
+        if a is None:
+            a = bulk(name=element_a).get_neighbors(num_neighbors=1).distances[0, 0] * (4 / np.sqrt(3))
```

Comment:
```suggestion
            a = bulk(name=element_a, cubic=True).cell[0, 0]
```
We could also look them up from ASE.ChemicalElements or mendeleev, but I'd have to look that up.

Reply:
> We could also look them up from ASE.ChemicalElements or mendeleev, but I'd have to look that up.

I think that's what the ASE creator is doing anyhow. IMO it's not enough of a performance hit for the more complicated.

I can't merge the commit though as it breaks in a few ways -- and makes a test fail, which is a good thing here! The base cell of C15 is diamond cubic, so what I'm actually doing is making sure the less-common species has the same nearest neighbour distance as it would in the bulk. I just checked, and using Fe it's off by a factor of two. For HCP the test breaks entirely because of `cubic=True` and is off by not quite a factor of two (I guess because of the c/a ratio).

Reply:
I'm open to just `2*bulk(name=element_a).cell[0, 0]` if that's a good enough approximation. I haven't played around relaxing these phases and have no intuition for how much they contract or expand relative to the pure bulk phases.